.cloudbuild/publish-image.sh

```diff
@@ -0,0 +1,7 @@
+#! /bin/env bash
+set -e
+IMAGE_NAME=$DOCKER_REGISTRY/$DOCKER_REPOSITORY:$DOCKER_TAG
+echo Building $IMAGE_NAME
+DOCKER_BUILDKIT=1 docker build -t $IMAGE_NAME .
+echo $QUAYIO_DOCKER_PASSWORD | docker login -u $QUAYIO_DOCKER_USERNAME --password-stdin $DOCKER_REGISTRY
```

Comment:
Full build log:
```
starting build "7eb2bed4-2503-4b56-a7ec-693002698192"

FETCHSOURCE
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change. To configure the initial branch name to use in all
hint: of your new repositories, which will suppress this warning, call:
hint: 
hint: 	git config --global init.defaultBranch <name>
hint: 
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint: 'development'. The just-created branch can be renamed via this command:
hint: 
hint: 	git branch -m <name>
Initialized empty Git repository in /workspace/.git/
From https://github.com/gravitational/docs
 * branch            f0dfc29fdd775bd814de5bee816990d4d4cb9d98 -> FETCH_HEAD
HEAD is now at f0dfc29 repo setup
SETUPBUILD
BUILD
Already have image (with digest): gcr.io/cloud-builders/docker
Building quay.io/gravitational/docs:dev
#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 124B done
#1 DONE 0.1s

#2 [internal] load .dockerignore
#2 transferring context: 2B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/library/node:14-slim
#3 DONE 0.5s

#4 [1/4] FROM docker.io/library/node:14-slim@sha256:8afd17e3e9cae76b76550c4...
#4 resolve docker.io/library/node:14-slim@sha256:8afd17e3e9cae76b76550c4f3bd9adfe8438bdedfdd4c9a27823669aa0f13750 done
#4 sha256:c1a47ef415ea319ef9a75bcb74c04c3b0b90fd0b2711fe2a000b4a0b06dfc091 1.37kB / 1.37kB done
#4 sha256:cbe52dc30fe81c912859c8dba6812cc29ac88a1b51076225883a337bf1fc0bef 6.99kB / 6.99kB done
#4 sha256:73e8e05762864dd5f0a339ca8471e27294d125b63c8adfb278e84c105b8c842d 0B / 22.57MB 0.1s
#4 sha256:99f62411a8479911bdbcf3b4c86890b69dc5163d395619a292c4d993a2bd5f4b 0B / 4.18kB 0.1s
#4 sha256:7bf5c484a709b9c2352edd8d878c49da8733b97fe1a14cbc42afcc783f10c0b1 0B / 36.11MB 0.1s
#4 sha256:8afd17e3e9cae76b76550c4f3bd9adfe8438bdedfdd4c9a27823669aa0f13750 776B / 776B done
#4 ...

#6 [internal] load build context
#6 transferring context: 2.89MB 0.1s done
#6 DONE 0.1s

#4 [1/4] FROM docker.io/library/node:14-slim@sha256:8afd17e3e9cae76b76550c4...
#4 sha256:73e8e05762864dd5f0a339ca8471e27294d125b63c8adfb278e84c105b8c842d 10.96MB / 22.57MB 0.2s
#4 sha256:99f62411a8479911bdbcf3b4c86890b69dc5163d395619a292c4d993a2bd5f4b 4.18kB / 4.18kB 0.1s done
#4 sha256:eba1cf6fbc561ebcd845311e19809d6d36ca7b2862c843f6529c86d27a80161a 0B / 2.79MB 0.2s
#4 sha256:73e8e05762864dd5f0a339ca8471e27294d125b63c8adfb278e84c105b8c842d 18.87MB / 22.57MB 0.3s
#4 sha256:7bf5c484a709b9c2352edd8d878c49da8733b97fe1a14cbc42afcc783f10c0b1 9.44MB / 36.11MB 0.3s
#4 sha256:eba1cf6fbc561ebcd845311e19809d6d36ca7b2862c843f6529c86d27a80161a 1.58MB / 2.79MB 0.3s
#4 sha256:73e8e05762864dd5f0a339ca8471e27294d125b63c8adfb278e84c105b8c842d 22.57MB / 22.57MB 0.4s
#4 sha256:7bf5c484a709b9c2352edd8d878c49da8733b97fe1a14cbc42afcc783f10c0b1 22.02MB / 36.11MB 0.4s
#4 sha256:eba1cf6fbc561ebcd845311e19809d6d36ca7b2862c843f6529c86d27a80161a 2.79MB / 2.79MB 0.3s done
#4 sha256:3973484d1d9c1d0d47675bb6c2a6fa29096a0d62af926047e4044c1efd3b8918 0B / 465B 0.4s
#4 sha256:7bf5c484a709b9c2352edd8d878c49da8733b97fe1a14cbc42afcc783f10c0b1 36.11MB / 36.11MB 0.5s
#4 sha256:3973484d1d9c1d0d47675bb6c2a6fa29096a0d62af926047e4044c1efd3b8918 465B / 465B 0.5s
#4 extracting sha256:73e8e05762864dd5f0a339ca8471e27294d125b63c8adfb278e84c105b8c842d
#4 sha256:73e8e05762864dd5f0a339ca8471e27294d125b63c8adfb278e84c105b8c842d 22.57MB / 22.57MB 0.5s done
#4 sha256:3973484d1d9c1d0d47675bb6c2a6fa29096a0d62af926047e4044c1efd3b8918 465B / 465B 0.6s done
#4 sha256:7bf5c484a709b9c2352edd8d878c49da8733b97fe1a14cbc42afcc783f10c0b1 36.11MB / 36.11MB 0.8s done
#4 extracting sha256:73e8e05762864dd5f0a339ca8471e27294d125b63c8adfb278e84c105b8c842d 1.4s done
#4 extracting sha256:99f62411a8479911bdbcf3b4c86890b69dc5163d395619a292c4d993a2bd5f4b done
#4 extracting sha256:7bf5c484a709b9c2352edd8d878c49da8733b97fe1a14cbc42afcc783f10c0b1
#4 extracting sha256:7bf5c484a709b9c2352edd8d878c49da8733b97fe1a14cbc42afcc783f10c0b1 2.2s done
#4 extracting sha256:eba1cf6fbc561ebcd845311e19809d6d36ca7b2862c843f6529c86d27a80161a 0.1s
#4 extracting sha256:eba1cf6fbc561ebcd845311e19809d6d36ca7b2862c843f6529c86d27a80161a 0.2s done
#4 extracting sha256:3973484d1d9c1d0d47675bb6c2a6fa29096a0d62af926047e4044c1efd3b8918
#4 extracting sha256:3973484d1d9c1d0d47675bb6c2a6fa29096a0d62af926047e4044c1efd3b8918 done
#4 DONE 5.0s

#5 [2/4] WORKDIR /src
#5 DONE 1.1s

#7 [3/4] COPY . /src
#7 DONE 0.1s

#8 [4/4] RUN yarn && yarn build-node
#8 0.512 yarn install v1.22.19
#8 0.657 [1/4] Resolving packages...
#8 1.492 [2/4] Fetching packages...
#8 46.65 [3/4] Linking dependencies...
#8 62.79 [4/4] Building fresh packages...
#8 64.91 Done in 64.41s.
#8 65.11 yarn run v1.22.19
#8 65.15 $ rm -rf .build && tsc --build tsconfig.node.json && tsc-esm-fix --target='.build' --ext='.mjs'
#8 91.19 Done in 26.08s.
#8 DONE 91.9s

#9 exporting to image
#9 exporting layers
#9 exporting layers 19.7s done
#9 writing image sha256:29faa0df30898cec7a395e8752fd583671f262a0307620ccbe0810eec9690fa9 done
#9 naming to quay.io/gravitational/docs:dev done
#9 DONE 19.7s
WARNING! Your password will be stored unencrypted in /builder/home/.docker/config.json.
Configure a credential helper to remove this warning. See
https://docs.docker.com/engine/reference/commandline/login/#credentials-store

Login Succeeded
docker push quay.io/gravitational/docs:dev
The push refers to repository [quay.io/gravitational/docs]
7d656d7944e1: Preparing
1123db2f1581: Preparing
85d1ed2ef781: Preparing
deaf1a29afc8: Preparing
eb814cd9fc2a: Preparing
63b9ca33cf62: Preparing
905f245524ab: Preparing
2ddc9f2aee6d: Preparing
63b9ca33cf62: Waiting
905f245524ab: Waiting
2ddc9f2aee6d: Waiting
deaf1a29afc8: Layer already exists
eb814cd9fc2a: Layer already exists
63b9ca33cf62: Layer already exists
905f245524ab: Layer already exists
2ddc9f2aee6d: Layer already exists
85d1ed2ef781: Pushed
1123db2f1581: Pushed
7d656d7944e1: Pushed
dev: digest: sha256:8f0d97339d2bc8408cb07acb879ac13413900b6efe794ab217ae9fa9ddc4e2b7 size: 1998
PUSH
DONE
```